src/main/java/org/folio/entlinks/service/dataloader/SourceFileDeserializer.java

```diff
@@ -44,14 +45,24 @@ public AuthoritySourceFile deserialize(JsonParser parser, DeserializationContext
     var type = Optional.ofNullable(node.get("type")).map(JsonNode::asText).orElse(null);
     var name = Optional.ofNullable(node.get("name")).map(JsonNode::asText).orElse(null);
     var url = Optional.ofNullable(node.get("baseUrl")).map(JsonNode::asText).orElse(null);
-    var source = Optional.ofNullable(node.get("source")).map(JsonNode::asText).orElse(null);
+    final var source = Optional.ofNullable(node.get("source")).map(JsonNode::asText).orElse(null);
 
     sourceFile.setId(id);
     sourceFile.setName(name);
     sourceFile.setType(type);
     sourceFile.setBaseUrl(url);
-    sourceFile.setSource(source);
-
+    setSourceType(sourceFile, source);
     return sourceFile;
   }
+
+  private static void setSourceType(AuthoritySourceFile sourceFile, String source) {
+    if (source != null) {
+      switch (source) {
+        case "LOCAL" -> sourceFile.setSource(AuthoritySourceFileSource.LOCAL);
+        case "FOLIO" -> sourceFile.setSource(AuthoritySourceFileSource.FOLIO);
+        case "CONSORTIUM" -> sourceFile.setSource(AuthoritySourceFileSource.CONSORTIUM);
```

Comment:
System Source Files can have only ```folio``` in source field, so you can omit checks for other values

Reply:
You still have a checks for "LOCAL" and "CONSORTIUM" sources, but it should be a case that we create reference-date for source files with these sources. We only create source files with source "FOLIO".